spyder/plugins/switcher/widgets/switcher.py

```diff
@@ -160,31 +158,38 @@ class Switcher(QDialog):
     def __init__(self, parent, help_text=None, item_styles=ITEM_STYLES,
                  item_separator_styles=ITEM_SEPARATOR_STYLES):
         """Multi purpose switcher."""
-        super(Switcher, self).__init__(parent)
+        super().__init__(parent)
+
+        # Attributes
         self._modes = {}
         self._mode_on = ''
         self._item_styles = item_styles
         self._item_separator_styles = item_separator_styles
+        self.current_project = None
+        self.projects_section = None
 
         # Widgets
-        self.timer = QTimer()
         self.edit = QLineEdit(self)
         self.list = QListView(self)
         self.model = QStandardItemModel(self.list)
         self.proxy = SwitcherProxyModel(self.list)
         self.filter = KeyPressFilter()
 
-        # Widgets setup
-        self.timer.setInterval(300)
-        self.timer.setSingleShot(True)
-        self.timer.timeout.connect(self.setup)
+        # Search timer
+        self._search_timer = QTimer(self)
+        self._search_timer.setInterval(300)
+        self._search_timer.setSingleShot(True)
+        self._search_timer.timeout.connect(self._on_search_text_changed)
 
+        # Widgets setup
         self.setWindowFlags(Qt.Popup | Qt.FramelessWindowHint)
         self.setWindowOpacity(0.95)
-#        self.setMinimumHeight(self._MIN_HEIGHT)
+        # self.setMinimumHeight(self._MIN_HEIGHT)
```

Comment:
Should we removed this line or if there is a reason to keep it maybe add the clarification as a comment above?

Reply:
I think we can leave it in case we need for later.